src/Component/src/Metadata/Resource/Factory/PhpFileResourceClassListFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Resource\Factory;
+
+use Sylius\Resource\Metadata\Extractor\ResourceExtractorInterface;
+use Sylius\Resource\Metadata\Resource\ResourceClassList;
+
+/**
+ * Creates a resource class list from PHP configuration files.
+ *
+ * @experimental
+ */
+final class PhpFileResourceClassListFactory implements ResourceClassListFactoryInterface
+{
+    public function __construct(
+        private readonly ResourceExtractorInterface $metadataExtractor,
```

Comment:
maybe `$resourceExtractor` or `$resourceMetadataExtractor`

Reply:
Since the whole point of the this class to call phpFileResourceMetataResourceExtractor, then would be cool to use in the name to understand it at glance without checking service definition.

```suggestion
        private readonly ResourceExtractorInterface $phpFileResourceMetadataExtractor,
```

Reply:
Very verbose, but ok.